common/djangoapps/student/views.py

```diff
@@ -1384,7 +1384,9 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
             user = User.objects.get(email=email)
         except User.DoesNotExist:
             try:
-                user = User.objects.get(username=email)
+                # try using the email as username
+                username = email
```

Comment:
I was just trying to make it clearer that maybe the email is actually the username :( but ok I'll try and think on something